wire-ios-utilities/Source/Public/NSManagedObjectContext+GroupQueue.swift

```diff
@@ -0,0 +1,140 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import CoreData
+import WireSystem
+
+extension NSManagedObjectContext: GroupQueue {
+
+    @objc
+    public var dispatchGroup: ZMSDispatchGroup? {
+        dispatchGroupContext?.groups.first
+    }
+
+    public func performGroupedBlock(_ block: @escaping () -> Void) {
+        let groups = dispatchGroupContext?.enterAll() ?? []
+        let timePoint = TimePoint(interval: PerformWarningTimeout)
+        perform {
+            timePoint.resetTime()
+            block()
+            self.dispatchGroupContext?.leave(groups)
+            timePoint.warnIfLongerThanInterval()
+        }
+    }
+}
+
+extension NSManagedObjectContext {
+
+    @objc
+    public var pendingSaveCounter: Int {
+        get { objc_getAssociatedObject(self, &AssociatedPendingSaveCountKey) as? Int ?? 0 }
+        set { objc_setAssociatedObject(self, &AssociatedPendingSaveCountKey, newValue, .OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+    }
```

Comment:
Maybe add a comment that we should refactor this code and not rely on `objc_setAssociatedObject` in the long term?